sos/policies/__init__.py

```diff
@@ -1402,7 +1402,7 @@ def pre_work(self):
                                            "that you are generating this "
                                            "report for [%s]: ") % caseid)
                 # Policies will need to handle the prompts for user information
-                if cmdline_opts.upload or self.upload_url:
+                if cmdline_opts.upload and self.get_upload_url():
```

Comment:
Why this change (esp. compared to "upload or upload_url" at https://github.com/sosreport/sos/pull/2210/files#diff-e07b4b5df8f32d4dab336b60233ae2aeR1130), please?

Otherwise nice extension.

Reply:
Hmm, the prompts should set the necessary bits to allow the latter addition to execute appropriately - but from a consistency standpoint you're absolutely right. I'll change the latter to also check `get_upload_url()`.